javascript/modules/client/src/internal/interfaces/plugins.ts

```diff
@@ -74,19 +80,29 @@ export interface IProposalSettings {
   minDuration: number;
 }
 
-export interface ICreateProposal {
+export interface ICreateProposalParams {
+  pluginAddress: string
   metadata: ProposalMetadata;
   actions?: DaoAction[];
   startDate?: Date;
   endDate?: Date;
   executeOnPass?: boolean;
   creatorVote?: VoteValues;
 }
+export interface IVoteProposalParams {
+  pluginAddress: string
+  vote: VoteValues;
+  proposalId: string;
+}
+export interface IExecuteProposalParams {
+  pluginAddress: string
+  proposalId: string;
+}
 
 // Factory init params
 
 export type IErc20PluginInstall = {
-  proposals: IProposalSettings;
+  proposals: IPluginSettings;
```

Comment:
Since the plugin settings can grow beyond the token scope, would you feel like `settings: IPluginSettings` makes more sense now?

Reply:
Yes, now it makes more sense in general

Reply:
Let's go for it